### ISSUE TITLE
Screen space reflections not working correctly

### ISSUE THREAD

**Issue by kurzemnieks** [Unity-Technologies/PostProcessing#368]
Screen space reflections in v2 on Unity 2017.1 and 2017.2 are not working correctly.
They always appear to be weirdly streched up until object mesh. I don't think this is how it is supposed to look - tried different settings but nothing changes. 
Is this a bug or it's supposed to look like that? 

![ssr_problem](https://user-images.githubusercontent.com/1818587/32979212-8fdaff18-cc59-11e7-9318-d71042252c4d.png)

**Comment by gokselgoktas** [comment]
Hi!

This happens because we don't (and really, cannot) know the volume thickness corresponding to each texel location on the buffer.

At some point during development we decided to get rid of the intermediate back-face camera depth render we were doing in order have an estimate for this, for performance reasons. Ever since then, a hit on the depth buffer, no matter at what depth, meant a hit. This in turn leads to what you're seeing on your screen.

For most use cases this works just fine, especially considering the performance gains obtained from not having to render the scene a second time and sample that texture at a critical part of the algorithm. Keep in mind that conceptually SSR itself is a bad choice for completely smooth reflections due to its many limitations.

To conclude, we're still looking into this, and we'll be keeping this in mind.
Thanks!

**Comment by kurzemnieks** [comment]
Thanks for the answer. 
I wonder how is this working in UE - it seems more correct there?!

**Comment by artyom-zuev** [comment]
Yeah, I find this artifact to be too noticeable with our scenes, it's making objects like I``-shaped street lights and trees turn into thick columns in reflections, which is very noticeable even with blurry reflections.

**Comment by oyvind-stromsvik** [comment]
This is still like this in the latest version of the post processing stack v.2.

I don't get this response:
>For most use cases this works just fine

Won't this be wrong like the provided images in every case where the objects reflected are without an opaque background?

I'm asking because I'm unable to produce a scene which doesn't have this artifact show up everywhere.

And I don't think I understand the thickness issue. Why do you need to know that? Can't we just reflect what the current camera is seeing? Wouldn't that be much more correct than what we're seeing here?

Lastly, in general I would like more options for the post processing stack. Even if these options are less than optimal or produce wrong or buggy results. Just hide them under an "advanced" settings tab imo so that those of us who want to enable or play with these settings can do so.

**Comment by hdachev** [comment]
I think this is the relevant bit from the rendering of Inside:
https://youtu.be/RdN06E6Xn9E?t=2666

I think this is what he's calling "the boy with MC Hammer pants", I always wondered what that was.

**Comment by artyom-zuev** [comment]
Here is an example of an SSR implementation I'd prefer to see in the package - as you can see in the sample screenshots, it has no issues with streaking under objects:

https://github.com/MaxwellGengYF/Unity-Screen-Space-Reflection

**Comment by RobbieOnGithub** [comment]
This needs to be fixed... completely unusable for most circumstances with this stretching effect.... I hope Unity adds a version of SSR that's looks decent. I'm fine with a performance hit. Maybe they can pay Playdead for their Inside code and give us that.

**Comment by zenberger** [comment]
totally agree with the complaints mentioned here. On top, in the manual, the screenshots from the Lab Corridor do not show these stretches. Could we get a setting that allows what Unity is using? Do not send us to hack this with reflection probes...
![screenshot 2019-02-27 at 03 10 38](https://user-images.githubusercontent.com/47144438/53462946-67c6ab00-3a3d-11e9-806a-b82461a99d99.png)
